### PR TITLE
Zadanie 2  + poprawilem zad1 ale tego nie trzeba reviewować

### PR DESCRIPTION
**Pull request by FaziSPB**

### CODE REVIEW

**Review by Jagaxxx — COMMENTED** [review]
Hej,

Ciekawe wykresy i trafne wnioski - fajnie, że zwizualizowałeś zależność accuracy od hiperparametrów. Widać, co dokładnie dzieje się z modelem przy overfittingu.

**Comment by JAANULO** [comment]
Ogólnie fajnie, analiza, obrazki , wnioski

Inną opcją do szukania parametrów też jest 
Optuma / Optymalizacja Bayesowska, która sama w sobie jest 'mądrzejsza' od grid Search, polecam zerknąć 

Możesz też się zastanowić się nad jakimś ograniczeniem treningu; 

Ilość iteracji - bo jak ustawisz bardzo dużo kombinacji to trening będzie bardzo trwał
Czas - albo limitujesz trening przez jakiś minutnik czy inny czasomierz

**Review by KotwinskiDiego — COMMENTED** [review]
Po pierwsze wydaje mi sie ze niepotrzebnie dodales sobie roboty najpierw dzielac dane zeby potem je osobno czyscic 
Widze ze jako metryki oceny uzyles accuracy, brakuje mi uzasadnienia dlaczego?
Po za tym nie mam wiecej zastrzeżeń

Praca generalnie wykonana prawidlowo,
Wizualizacja accuracy wzgledem hipermarametrow byla swietnym pomyslem, bardzo mi sie podoba.
Podoba mi sie rowniez stworzenie heatmap na koncu oraz wyciagniete z nich wnioski.

**Comment by FaziSPB** [comment]
> Po pierwsze wydaje mi sie ze niepotrzebnie dodales sobie roboty najpierw dzielac dane zeby potem je osobno czyscic  

Wydaje mi się, że gdybym najpierw je wyczyścił a potem podzielił to doszłoby do wycieku danych. Przed podziałem na zbiór treningowy i testowy można przeprowadzać tylko pomniejsze zmiany jak np. redukcja kolumn. Czyszczenie, kodowanie oraz skalowanie musi odbyć się już po podziale, dlatego na secie treningowym używam fit_transform a na testowym tylko transform, ponieważ uczymy model na zbiorze treningowym, a w zbiór testowy wstrzykujemy wartości zapamiętane na zbiorze treningowym, żeby - jak wyżej, uniknąć wycieku danych.

**Comment by DamianRyczko** [comment]
Hej, fajnie wykonane zadanie - grid search po parametrach, kroswalidacja i pipeline'y na duży plus.

Kilka uwag (niekoniecznie do poprawy, choć zachęcam do pochwalenia się wynikiem po ich uwzględnieniu):

1. https://scikit-learn.org/stable/modules/generated/sklearn.compose.ColumnTransformer.html
Warto pamiętać, że ColumnTransformer domyślnie robi drop na kolumnach, które nie zostały zdefiniowane w transformerach. Niestety przez to Twój feature engineering nigdy nie dotarł do modelu.

2. Zauważyłeś, że drzewo radzi sobie najlepiej przy głębokości od 7 do 14. Wcześniej jednak napisałeś, że DecisionTree potrzebował maksymalnej możliwej głębokości (zakładam, że chodzi o 10 z siatki parametrów). Może warto było wrócić do gridsearcha i pozwolić mu na 14 albo więcej - wynik mógł się poprawić.

3. Fajnie, że zauważyłeś, iż drzewo decyzyjne ma swoje wady i zalety. Swoją drogą, możesz zajrzeć, jak to drzewo wygląda: 'from sklearn.tree import plot_tree, export_text'. Skoro jedno drzewo radzi sobie dobrze, można było pokusić się o wnioski, jak poradziłby sobie las losowy albo model typu XGBoost, oparty o boostowane drzewa (ciekawostka, nie wymagam tego).

zadanie zaliczone

**Review by DamianRyczko — APPROVED** [review]
